parser/healthchecks.go

```diff
@@ -0,0 +1,245 @@
+package parser
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/deis/workflow-cli/cmd"
+
+	"github.com/deis/controller-sdk-go/api"
+	docopt "github.com/docopt/docopt-go"
+)
+
+// Healthchecks routes ealthcheck commands to their specific function
+func Healthchecks(argv []string) error {
+	usage := `
+Valid commands for healthchecks:
+
+healthchecks:list        list healthchecks for an app
+healthchecks:set         set healthchecks for an app
+healthchecks:unset       unset healthchecks for an app
+
+Use 'deis help [command]' to learn more.
+`
+
+	switch argv[0] {
+	case "healthchecks:list":
+		return healthchecksList(argv)
+	case "healthchecks:set":
+		return healthchecksSet(argv)
+	case "healthchecks:unset":
+		return healthchecksUnset(argv)
+	default:
+		if printHelp(argv, usage) {
+			return nil
+		}
+
+		if argv[0] == "healthchecks" {
+			argv[0] = "healthchecks:list"
+			return healthchecksList(argv)
+		}
+
+		PrintUsage()
+		return nil
+	}
+}
+
+func healthchecksList(argv []string) error {
+	usage := `
+Lists healthchecks for an application.
+
+Usage: deis healthchecks:list [options]
+
+Options:
+  -a --app=<app>
+    the uniquely identifiable name of the application.
+`
+
+	args, err := docopt.Parse(usage, argv, true, "", false, true)
+
+	if err != nil {
+		return err
+	}
+
+	return cmd.HealthchecksList(safeGetValue(args, "--app"))
+}
+
+func healthchecksSet(argv []string) error {
+	usage := `
+Sets healthchecks for an application.
+
+By default, Workflow only checks that the application starts in their Pod. A health check
+may be added by configuring a health check probe for the application. The health checks
+are implemented as Kubernetes Pod Probes. A 'liveness' and a 'readiness' probe can be
+configured, and each probe can be of type 'httpGet' or 'exec' depending on the type of
+probe the Pod requires.
+
+A 'liveness' probe is useful for applications running for long periods of time, eventually
+transitioning to broken states and cannot recover except by restarting them.
+
+Other times, a 'readiness' probe is useful when the Pod is only temporarily unable to
+serve, and will recover on its own. In this case, if a Pod fails its 'readiness' probe,
+the Pod will not be shut down, but rather the Pod will stop receiving incoming requests.
+
+'httpGet' probes are just as it sounds: it performs a HTTP GET operation on the Pod. A
+response code inside the 200-399 range is considered a pass. 'httpGet' probes accept a
+port number to perform the HTTP GET operation on the Pod.
```

Comment:
Should call out the path here, and make it explicit (like the docs) that only path is supported, no query params or anchor, or full URL